contents/handbook/engineering/development-process.md

```diff
@@ -57,55 +30,112 @@ After the sprint planning, each team should have their own session (can be async
 * **Challenging:** It should too big for one person to solve on their own and require creativity or brute force to achieve in the proposed time-frame (e.g. ship correlation analysis with a killer feature no one else has)
 * **Homogenous:** The goal should be all about achieving a single meaningful thing and not a collection of unconnected things (i.e. NOT ‘Improve query performance and launch collaboration MVP’)
 
-Goals should be time-bound, but since we primarily use goals for our two-weekly sprint planning we should consider them generally timebound to two weeks.
+### Assign an owner
+
+A single engineer should be accountable for a milestone partnering closely other functions to ensure it’s successful.
+
+### Think about other teams
+
+Most things won't cause issues for other teams. However, if it will - don't "align teams" or write a huge RFC (unless that'd help you). Do a quick 1/1 with the most relevant person on another team. 
+
+Consider:
+* The scale of the customer you're building for
+* If you can get from your hacky MVP to production-ready easily. It's OK to start with basic, but be mindful of making it harder to fully roll something out in future.
+* If you know what you're doing or need someone from another team's expertise to get the right architecture or overall approach. We have lots of experienced people, get their help if you would benefit from it.
+
+If this is a big feature which will need an announcement, content, or other marketing support then it's _never_ too early for [the owner](/handbook/engineering/development-process#assign-an-owner) to let the Marketing team know. Drop a post in their Slack channel or tagging them on an issue. 
+
+### Break up goals
+
+The owner turns the ambiguous milestone into a roadmap of ambitious, meaningful, sprint-sized goals, thinking 2 - 3 sprints ahead to give other functions time. [Goal principles](/handbook/engineering/development-process#set-some-milestones) still apply.
+
+### Execute on goals
+
+Our sprint planning session is where the high-level goals for the next sprint get decided. Each team takes these, breaks them down, and adds their own smaller items, like bugfixes.
+
+Sprint planning works like this:
+* During the week of the sprint planning, teams (with the help of the product managers) define one or two big ambitious goals for this sprint that align with major milestones.
+* On Wednesday we get together. We use breakout rooms to split into the small teams, and each team finalizes the ambitious goals that they want to tackle.
+* We come back in the same room, where
+* We do a quick retrospective on the goals for the last sprint (did we get everything done? Should we have prioritized other things?)
+* Each team lead presents the main goals for their team. We have an open discussion about whether the goals are the right goals for this sprint and whether they are ambitious enough.
```

Comment:
We haven't done that in a long time, teams just do their planning as part of their own meetings (or within the Wednesday standup), then the engineering-wide meeting is just a way of collecting the goals of all teams.
So of these four points, only the "Each team lead presents" one is up to date.

Reply:
I'm not deeply familiar with Engineering team processes, so I'm going to defer to Engineering team to keep this aspect up to date tbh.